agent/local/state.go

```diff
@@ -1106,6 +1115,13 @@ func (l *State) updateSyncState() error {
 					m[k] = v
 				}
 			}
+			if !wasCopied {
+				// Make a shallow copy since we're going to mutate it and other
+				// readers may be reading it and we want to avoid a race.
+				wasCopied = true
```

Comment:
Is this value ever read after this point? I'm not seeing a loop around this, but maybe I'm missing it?

Can we remove this line?

```suggestion
```